samples/boards/reel_board/mesh_badge/src/mesh.c

```diff
@@ -76,7 +76,94 @@ BT_MESH_HB_CB_DEFINE(hb_cb) = {
 	.recv = heartbeat,
 };
 
+static void show_mesh_status(bool mesh_started_ok, int addr)
+{
+	if (!mesh_started_ok) {
+		board_show_text("Starting Mesh Failed", false, K_SECONDS(2));
+	} else {
+		char buf[32];
+
+		snprintk(buf, sizeof(buf), "Mesh Started\nAddr: 0x%04x", addr);
+		board_show_text(buf, false, K_SECONDS(4));
+	}
+}
+
+static void cfg_cli_app_key_status_cb(struct bt_mesh_cfg_cli *cli, uint16_t
+				addr, uint8_t status, uint16_t net_idx,
+				uint16_t app_idx)
+{
+	struct bt_mesh_cfg_cli_mod_pub pub = {
+		.addr = GROUP_ADDR,
+		.app_idx = APP_IDX,
+		.ttl = DEFAULT_TTL,
+		.period = BT_MESH_PUB_PERIOD_SEC(10),
+	};
+	int err;
+
+	/* Bind to vendor model */
+	err = bt_mesh_cfg_cli_mod_app_bind_vnd(NET_IDX, addr, addr, APP_IDX,
+				MOD_LF, BT_COMP_ID_LF, NULL);
+	if (err) {
+		printk("Failed to bind to vendor model (err %d)\n", err);
+		err = false;
+		goto end;
+	}
+
+	err = bt_mesh_cfg_cli_mod_app_bind(NET_IDX, addr, addr, APP_IDX,
+				BT_MESH_MODEL_ID_GEN_ONOFF_SRV, NULL);
+	if (err) {
+		printk("Failed to bind to vendor model (err %d)\n", err);
+		err = false;
+		goto end;
+	}
+
+	err = bt_mesh_cfg_cli_mod_app_bind(NET_IDX, addr, addr, APP_IDX,
+				BT_MESH_MODEL_ID_SENSOR_SRV, NULL);
+	if (err) {
+		printk("Failed to bind to vendor model (err %d)\n", err);
+		err = false;
+		goto end;
+	}
+
+	/* Bind to Health model */
+	err = bt_mesh_cfg_cli_mod_app_bind(NET_IDX, addr, addr, APP_IDX,
+				BT_MESH_MODEL_ID_HEALTH_SRV, NULL);
+	if (err) {
+		printk("Failed to bind to Health Server (err %d)\n", err);
+		err = false;
+		goto end;
+	}
+
+	/* Add model subscription */
+	err = bt_mesh_cfg_cli_mod_sub_add_vnd(NET_IDX, addr, addr, GROUP_ADDR,
+				MOD_LF, BT_COMP_ID_LF, NULL);
+	if (err) {
+		printk("Failed to add subscription (err %d)\n", err);
+		err = false;
+		goto end;
+	}
+
+	err = bt_mesh_cfg_cli_mod_pub_set_vnd(NET_IDX, addr, addr, MOD_LF,
+				BT_COMP_ID_LF, &pub, NULL);
+	if (err) {
+		printk("Failed to set publication (err %d)\n", err);
+		err = false;
```

Comment:
No need to change `err`, `show_mesh_status` can be called like this: `show_mesh_status(err == 0, addr)`.

Reply:
Oh, right, that is easier, will change.